examples/light-switch-app/efr32/src/binding-handler.cpp

```diff
@@ -342,6 +401,31 @@ void SwitchWorkerFunction(intptr_t context)
     Platform::Delete(data);
 }
 
+void BindingWorkerFunction(intptr_t context)
+{
+    VerifyOrReturn(context != 0, ChipLogError(NotSpecified, "BindingWorkerFunction - Invalid work data"));
+
+    EmberBindingTableEntry * entry = reinterpret_cast<EmberBindingTableEntry *>(context);
+
+    if (entry->type == EMBER_UNICAST_BINDING)
+    {
+        CHIP_ERROR err = BindingManager::GetInstance().UnicastBindingCreated(entry->fabricIndex, entry->nodeId);
+        if (err != CHIP_NO_ERROR)
+        {
+            // Unicast connection failure can happen if peer is offline. We'll retry connection on-demand.
+            ChipLogProgress(NotSpecified,
+                            "Binding: Failed to create session for unicast binding to device " ChipLogFormatX64
+                            ": %" CHIP_ERROR_FORMAT,
+                            ChipLogValueX64(entry->nodeId), err.Format());
+        }
+    }
+
+    BindingTable::GetInstance().Add(*entry);
```

Comment:
Shouldn't we have a single "add" function that does both this and the two notifications, called from both here and the cluster?  Having every bindings consumer have to duplicate all this logic does not make any sense.

Reply:
done